kong/kong.lua

```diff
@@ -252,6 +252,18 @@ function Kong.balancer()
   end
 end
 
+function Kong.rewrite()
+  core.rewrite.before()
+
+  for _, plugin in ipairs(singletons.loaded_plugins) do
+    if plugin.schema.no_consumer and plugin.schema.no_api then
+      plugin.handler:rewrite()
+    end
+  end
+
+  core.rewrite.after()
+end
+
```

Comment:
> ie, the access() for every plugin is invoked regardless if the plugin implements that method

That's not what I mean. Because that access handler is only called if the user explcitly added that plugin through the management api.

In the code you have now, it will run the handler, even if the user did not configure it. Just a plugin "being in the system" will make it being executed.